### PR TITLE
Fix podman pod create --infra-command and --infra-image

### PR DESCRIPTION
**Pull request by rhatdan**
Currently infr-command and --infra-image commands are ignored
from the user.  This PR instruments them and adds tests for
each combination.

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
Fixes https://github.com/containers/podman/issues/7167

File: libpod/pod.go [hunk 1]
```diff
@@ -70,6 +70,13 @@ type PodConfig struct {
 
 	// ID of the pod's lock
 	LockID uint32 `json:"lockID"`
+
+	// InfraImage is the image a pod infra container will use to manage
+	// namespaces.
+	InfraImage string `json:"InfraImage,omitempty"`
```
Comment (mheon) [inline libpod/pod.go:76]: Throw these in InfraContainerConfig

**Comment by TomSweeneyRedHat** [comment]
You've some lint unhappiness in here @rhatdan

File: libpod/pod.go [hunk 2]
```diff
@@ -70,6 +70,13 @@ type PodConfig struct {
 
 	// ID of the pod's lock
 	LockID uint32 `json:"lockID"`
+
+	// InfraImage is the image a pod infra container will use to manage
+	// namespaces.
+	InfraImage string `json:"infraImage,omitempty"`
```
Comment (mheon) [inline libpod/pod.go:76]: Can you take these out of PodConfig now that they're in InfraContainerConfig?

Comment (rhatdan) [reply libpod/pod.go:76]: Light dawns on marble head.  Fixed.

**Comment by TomSweeneyRedHat** [comment]
Still fighting test issues @rhatdan

**Comment by rhatdan** [comment]
This is ready to go in,
@TomSweeneyRedHat @mheon @vrothberg @giuseppe @ashley-cui @QiWang19 @jwhonce @baude PTAL

File: cmd/podman/pods/create.go [hunk 3]
```diff
@@ -104,6 +103,18 @@ func create(cmd *cobra.Command, args []string) error {
 		createOptions.Share = nil
 	} else {
 		createOptions.Share = strings.Split(share, ",")
+		if cmd.Flag("infra-command").Changed {
+			createOptions.InfraCommand, err = cmd.Flags().GetString("infra-command")
```
Comment (vrothberg) [inline cmd/podman/pods/create.go:107]: Why is that necessary? It seems rather exotic. A comment would be very helpful.

Comment (vrothberg) [reply cmd/podman/pods/create.go:107]: Same below.

**Review by vrothberg — COMMENTED** [review]
Other than the nit, LGTM

File: cmd/podman/pods/create.go (hunk 3 above)
Comment (rhatdan) [reply cmd/podman/pods/create.go:107]: Not sure it is exotic, since this type of code is all over the cli.  Basically we want to make sure the user made a change using the CLI, rather then sending the default settings over to the server.  That way the server can differentiate if the user wanted a change or just to take defaults.

Comment (vrothberg) [reply cmd/podman/pods/create.go:108]: So we wouldn't take the defaults from the client containers.conf but from the server?

Comment (rhatdan) [reply cmd/podman/pods/create.go:108]: Yes

Comment (vrothberg) [reply cmd/podman/pods/create.go:108]: What's the reasoning for that? Do we document that somewhere?

I think there's some inconsistency among the commands w.r.t. to which defaults come from the client and server.

I am not against the idea but just want to understand if we have some policy for that decision.

Comment (rhatdan) [reply cmd/podman/pods/create.go:108]: All defaults should come from the server.  Any defaults that come from the client are a bug.  But I agree there are some coming from the client.

Comment (rhatdan) [reply cmd/podman/pods/create.go:108]: @vrothberg I now have comments, PTAL

**Review by TomSweeneyRedHat — COMMENTED** [review]
LGTM
